docs/src/manual/constraints.md

```diff
@@ -0,0 +1,40 @@
+```@meta
+DocTestSetup = :(using Ferrite)
+```
+
+# Constraints
+
+PDEs can in general be subjected to a number of constraints, 
+
+
+```math
+g_I(\boldsymbol a) = 0, \quad I = 1 \text{ to } n_c
+```
+
+where $g$ are (non-linear) constraint equations and $\boldsymbol{a}$ is a vector of the degrees 
+of freedom, and $n_c$ is the number of constraints. There are many ways to enforce these constraints, 
+e.g. penalty methods, Lagrange multiplier methods. A special case is when the constraint equations
+are linear, in which they can be enforced in a special way. This is explained below. 
+
+## Linear constraints
+When the constraints are linear, they can be written as 
+
+```math
+a_i = \sum_j C_{ij} a_j + g_i
+```
```

Comment:
I did not really know the difference between affine and linear constraints, so I had too google it... but they seems like affine constraints are linear constraints with added inhomogenites? 
I need affine constraints in my code, thats why I implemented it :)

Reply:
Okay :) Should we name them "affine" instead of "linear" then to avoid confusion?

Reply:
I think affine could be more confusing since I never heard it been used in a FE setting. Even though affine is the correct name, I think we should keep the name linear. But maybe it is best to vote on it :P

Reply:
https://www.dealii.org/current/doxygen/deal.II/affine__constraints_8h_source.html here is one example :P